ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixParticipant.java

```diff
@@ -451,11 +451,12 @@ private void awaitDisablingPartition() throws InterruptedException {
         CommonUtils.createHelixPropertyStore(zkConnectStr, propertyStoreConfig, null);
     String path = PARTITION_DISABLED_ZNODE_PATH + instanceName;
     int count = 1;
-    while (!helixPropertyStore.exists(path, AccessOption.PERSISTENT)) {
+    while (helixPropertyStore.exists(path, AccessOption.PERSISTENT)) {
       // Thread.sleep() pauses the current thread but does not release any locks
       Thread.sleep(clusterMapConfig.clustermapRetryDisablePartitionCompletionBackoffMs);
       logger.info("{} th attempt on checking the completion of disabling partition.", ++count);
     }
+    helixPropertyStore.close();
```

Comment:
one weird thing I noticed was that the stop() method in ZkCacheBaseDataAccessor also stops the background event handling thread in addition to shutting down the zkClient like in the close() method. We may want to call stop() instead

Reply:
sure, let me update the method.